plugins/janus_videoroom.c

```diff
@@ -2997,10 +2998,13 @@ static int janus_videoroom_access_room(json_t *root, gboolean check_modify, gboo
 		/* signed tokens bypass pin validation */
 		json_t *token = json_object_get(root, "token");
 		if(token) {
-			char room_descriptor[26];
+			char room_descriptor[100];
 			g_snprintf(room_descriptor, sizeof(room_descriptor), "room=%s", room_id_str);
-			if(gateway->auth_signature_contains(&janus_videoroom_plugin, json_string_value(token), room_descriptor))
-				return 0;
+			if(!gateway->auth_signature_contains(&janus_videoroom_plugin, json_string_value(token), room_descriptor)) {
+				g_snprintf(error_cause, error_cause_size, "Unauthorized (wrong token %s)", token);
```

Comment:
This is missing the `if(error_cause)` the other error management lines of code have, as `error_cause` may be `NULL`.

Reply:
done